File: src/main/java/com/citasmedicas/appcitasmedicas/Service/OfficeService.java
package com.citasmedicas.appcitasmedicas.Service;

import com.citasmedicas.appcitasmedicas.dto.Request.CreateOfficeRequest;
import com.citasmedicas.appcitasmedicas.dto.Request.UpdateOfficeRequest;
import com.citasmedicas.appcitasmedicas.dto.Response.OfficeResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface OfficeService {
    
    OfficeResponse create(CreateOfficeRequest req);
    
    OfficeResponse update(Long id, UpdateOfficeRequest req);
    
    OfficeResponse findById(Long id);
    
    Page<OfficeResponse> findAll(Pageable pageable);
    
    void delete(Long id);
}

File: frontend/src/services/officeService.js
import api from './api';

export const officeService = {
  getAll: (page = 0, size = 100) =>
    api.get(`/offices?page=${page}&size=${size}`),

  create: (data) =>
    api.post('/offices', data),

  update: (id, data) =>
    api.patch(`/offices/${id}`, data),

  delete: (id) =>
    api.delete(`/offices/${id}`),
};